doc/README.md

```diff
@@ -36,20 +36,19 @@ Official documentation
 ----------------------
 
 The source for the official documentation for Apache Cassandra can be found in
-the `source` subdirectory. The documentation uses [sphinx](http://www.sphinx-doc.org/)
-and is thus written in [reStructuredText](http://docutils.sourceforge.net/rst.html).
+the `source` subdirectory. The documentation uses [antora](http://www.antora.org/)
+and is thus written in [asciidoc](http://asciidoc.org).
+
+To build the HTML documentation, you will need to first install antora and the
+[Apache Cassandra theme](https://???). To add search, [lunr](https://lunrjs.com) must be installed.
 
-To build the HTML documentation, you will need to first install sphinx and the
-[sphinx ReadTheDocs theme](https://pypi.org/project/sphinx_rtd_theme/).
-When using Python 3.6 on Windows, use `py -m pip install sphinx sphinx_rtd_theme`, on unix
-use:
 ```
-pip install sphinx sphinx_rtd_theme
+npm i -g @antora/cli @antora/site-generator-default
+npm install lunr
 ```
 
 The documentation can then be built from this directory by calling `make html`
-(or `make.bat html` on windows). Alternatively, the top-level `ant gen-doc`
-target can be used.  When using Python 3.6 on Windows, use `sphinx_build -b html source build`.
+Alternatively, the top-level `ant gen-doc` target can be used.  
```

Comment:
`ant gen-doc` no longer works. (because `doc/Makefile` is removed)

There now now way to generate docs from the cassandra directory (everything is now done [here](https://github.com/apache/cassandra-website/blob/trunk/site-content/docker-entrypoint.sh#L23-L139)). 

Is there a minimum `doc/Makefile` we can put in place so that `ant gen-doc` runs in a basic form, for the sake of catching breakages…?

Reply:
see
https://github.com/polandll/cassandra/pull/16
and
https://github.com/apache/cassandra-website/pull/86

Reply:
Good catch. I guess we have a Makefile that does part of what the `generate_cassandra_versioned_docs` function in the [docker-entrypoint.sh](https://github.com/apache/cassandra-website/blob/trunk/site-content/docker-entrypoint.sh#L23-L139) does. That is, build the jar files, then call the nodetool and cassandra configuration adoc generation scripts. This would be the most that `gen-doc` could do and should be enough to weed out any breakages. The README also needs to be updated. We would need to make this change on all three branches.

As a bonus, we could probably update the `generate_cassandra_versioned_docs` function to call `ant gen-doc`.

Reply:
Good catch. I guess we have a Makefile that does part of what the `generate_cassandra_versioned_docs` function in the [docker-entrypoint.sh](https://github.com/apache/cassandra-website/blob/trunk/site-content/docker-entrypoint.sh#L23-L139) does. That is, build the jar files, then call the nodetool and cassandra configuration adoc generation scripts. This would be the most that `gen-doc` could do and should be enough to weed out any breakages. The README also needs to be updated. We would need to make this change on all three branches.

As a bonus, we could probably update the `generate_cassandra_versioned_docs` function to call `ant gen-doc`.

Reply:
Yup, covered in the PRs above.

But… i'm going to change the Makefile and ant targets for compatibility reasons.

Generating in-tree docs: `ant gen-doc` and `make html` are for a full antora build of the in-tree docs. This won't be implemented in these PRs, but will be reserved for when it is. (We had previously spoken about being able to create non-styled antora docs, even something that can be put into `man` pages.)

New targets for generating the dynamic adoc pages: `ant gen-asciidoc` and `make gen-asciidoc`. 
The former will be called from the `generate_cassandra_versioned_docs` function.

Reply:
PRs updated accordingly. (#16 needs to be applied to three branches.)

Reply:
There's another commit in https://github.com/polandll/cassandra/pull/16 (i botched the squash me commit you merged in already)

Reply:
PR for `cassandra-4.0` https://github.com/polandll/cassandra/pull/17 

PR for `cassandra-3.11` https://github.com/polandll/cassandra/pull/19